File: Franchise/get-pdf.pl
#!/usr/bin/perl -w

use strict;
use CGI;
use CGI::Carp qw ( fatalsToBrowser );
use File::Basename;

$CGI::POST_MAX = 1024 * 5000;
my $safe_filename_characters = "a-zA-Z0-9_.-";
my $upload_dir = "$ARGV[0]";

my $query = new CGI;
my $filename = $query->param("pdffile");

if ( !$filename )
{
 print "There was a problem uploading your file.";
 exit;
}

my ( $name, $path, $extension ) = fileparse ( $filename, '\..*' );
$filename = $name . $extension;
$filename =~ tr/ /_/;
$filename =~ s/[^$safe_filename_characters]//g;

if ( $filename =~ /^([$safe_filename_characters]+)$/ )
{
 $filename = $1;
}
else
{
 die "Filename contains invalid characters";
}

my $upload_filehandle = $query->upload("pdffile");

open ( UPLOADFILE, ">$upload_dir/$filename" ) or die "$!";
binmode UPLOADFILE;

while ( <$upload_filehandle> )
{
 print UPLOADFILE;
}

close UPLOADFILE;

print $filename;

File: Franchise/get-pdf.py
#!/usr/bin/python

import cgi, os, sys
import cgitb; cgitb.enable()

form = cgi.FieldStorage()
fileitem = form['pdffile']

if fileitem.filename:
   fn = sys.argv[1] + '/' + os.path.basename(fileitem.filename)
   fn = fn.replace(' ', '_')
   open(fn, 'wb').write(fileitem.file.read())
else:
   fn = ""

print(os.path.basename(fn))
